sysutils/terragrunt/Portfile

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           golang 1.0
+
+name                terragrunt
+version             0.23.4
+depends_run         port:terraform-0.12
+
+categories          sysutils
+platforms           darwin linux
+license             MIT
+maintainers         mjrc.nl:macports
+
+description         Terragrunt
+long_description    Terragrunt is a thin wrapper for Terraform that provides extra tools for working with multiple Terraform modules.
+homepage            https://terragrunt.gruntwork.io
+
+go.setup            github.com/gruntwork-io/terragrunt ${version} v
+
+checksums           rmd160  f072990cea4a0d0d2a814e32ab233ec64a93825e \
+                    sha256  3b4ddc07684eba41e1d3ad8b33464e8e9aeea6d2adb29aab6799a00f48988a52 \
+                    size    1963542
+
+destroot {
+    xinstall -m 755 ${worksrcpath}/${name} ${destroot}${prefix}/bin/
```

Comment:
the `terraform-XY` ports install their executable as `terraformXY`, I suspect that `terragrunt` is assuming the file to be called `terraform`- correct? Is there anything  you can do about specifying this during the build? Alternatively, it appears there if a `terragrunt-tfpath` environment variable that could potentially work (if it allows to specify the full path); if that's the case notifying the end user (i.e., with a message in `notes`) would probably be good.

Reply:
Correct, and thank you for pointing this out. I've tested and I've found out that `terraform` is implemented as a link to the binary as shown below, which seems to be a working fine. 

`0 lrwxr-xr-x  1 root  wheel  28 Apr  2 20:51 /opt/local/bin/terraform -> /opt/local/bin/terraform0.12
`

Also, the variable you mentioned [does indeed exist](https://terragrunt.gruntwork.io/docs/reference/cli-options/#terragrunt-tfpath). Would it then suffice to only add a note, pointing out the option to adjust the `terraform` location? As I'm not seeing what else exactly would be necessary.

Reply:
@mjrc that statement is not correct. The `terraform-0.12` port only installs the following files:
```
> port contents terraform-0.12
Port terraform-0.12 contains:
  /opt/local/bin/terraform0.12
  /opt/local/etc/select/terraform/terraform0.12
```

you have probably run `sudo port select --set terraform terraform0.12` as that would indeed make the link you're referring to. However, you can and should not rely on the fact that an end user has done the same. 

I think you should first test whether you can actually use the `/opt/local/bin/terraform0.12` file by using the variable; if that works you could get away with just adding a note to inform the user of this. Ideally, you would patch the build system (if possible) to use the `terraform0.12` binary instead so that it will work without going through additional steps. 

ping @herbygillot as the maintainer of `terraform` - any suggestions here?

Reply:
After taking a quick look, as far as I can tell, options are limited.  It looks like we will have to  notify the user either way.

I say so because it looks like `terragrunt` unfortunately does not let us have a system-wide configuration file, which if that were an option, could be used to configure `terragrunt` to always use `/opt/local/bin/terraform0.12`.

So the options you've already discussed are the only ways to go as far as I can, and the user will have to be informed of either, whether it's:

**Option A**: ensuring Terraform 0.12 is currently the selected version by way of `port select`

**Option B**: ensuring that Terragrunt is pointing to `/opt/local/bin/terraform0.12` via the `TERRAGRUNT_TFPATH` environment variable, or `--terragrunt-tfpath` parameter.

I would personally go with **A**, and simply show the user a note informing them that Terragrunt will require Terraform 0.12 to be currently selected in order to work, and inform the user that this can be done with `sudo port select --set terraform terraform0.12`.

Just my 2¢